contracts/cw721-sellable/src/execute.rs

```diff
@@ -101,6 +100,54 @@ pub fn try_list(
     Ok(Response::new().add_attribute("method", "list"))
 }
 
+pub fn try_redeem(
+    deps: DepsMut,
+    info: MessageInfo,
+    address: String,
+    ticket_id: &String,
+) -> Result<Response, ContractError> {
+    let contract = Cw721SellableContract::default();
+
+    // Validate only contract owner can call method
+    let minter = contract.minter.load(deps.storage)?;
+    if info.sender != minter {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    // Load ticket, error if ticket does not exist
+    let mut ticket = contract.tokens.load(deps.storage, ticket_id)?;
+
+    // Make sure owner param matches ticket owner
+    if ticket.owner != address {
+        return Err(ContractError::Unauthorized);
+    }
+
+    // Make sure ticket isn't locked or redeemed
+    match ticket.extension {
+        Some(ref metadata) => {
+            if (*metadata).redeemed.unwrap_or(false) {
```

Comment:
Rust is nice to us so we don't actually have to explicitly Derek

```suggestion
            if metadata.redeemed.unwrap_or(false) {
```